packages/google_maps_flutter/google_maps_flutter_platform_interface/lib/src/types/cluster_manager.dart

```diff
@@ -0,0 +1,85 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/foundation.dart' show immutable;
+import 'types.dart';
+
+/// Uniquely identifies a [ClusterManager] among [GoogleMap] clusters.
+///
+/// This does not have to be globally unique, only unique among the list.
+@immutable
+class ClusterManagerId extends MapsObjectId<ClusterManager> {
+  /// Creates an immutable identifier for a [ClusterManager].
+  const ClusterManagerId(String value) : super(value);
+}
+
+/// [ClusterManager] manages marker clustering for set of [Marker]s that have
+/// the same [ClusterManagerId] set.
```

Comment:
What is the conceptual difference between a `Cluster` and a `ClusterManager`? Why do we need two objects for clusters rather than one?

Reply:
`ClusterManager` is dart presentation of the native `ClusterManager` object that controls clustering for the markers given for it. In most of the cases developers need only one cluster manager, but in some cases having multiple could be needed. 
`Cluster` is one set of markers belonging to one cluster set on the map. `ClusterManager`s internal algorithms decides which markers belong to which cluster and if these markers are presented as they are as individual markers or as combined cluster marker (that marker with number inside of it).

Having possibility to handle multiple ClusterManagers allows developers to have separate `onClusterTap` functionality for different marker categories, and also gives more customization options later.

One possible use case could be to keep markers inside certain areas inside own clusterManagers to avoid them being clustered to same clusters. And when zoomed out they could represent correct amount of markers inside each area/country. This cannot be done with only one ClusterManager without having support for custom algorithms.

We could have had only boolean flag 'cluster' for markers to enable clustering for them, and have automatically one ClusterManager for each platform, but this then API limits what developers can do with the clustering, specially if new features are implemented giving options to add infoWindow for clusters, or change clustericon for clusters, created by certain cluster manager.

Reply:
Makes sense, thanks for the context!